packages/x-components/src/components/panels/base-tabs-panel.vue

```diff
@@ -0,0 +1,470 @@
+            -->
+          <slot name="tab-content" v-bind="{ tab, isSelected: tabIsSelected(tab) }">
+            {{ tab }}
+          </slot>
+        </button>
+      </slot>
+    </component>
+
+    <component :is="contentAnimation">
+      <div
+        v-if="selectedTab && $scopedSlots[selectedTab]"
+        :key="selectedTab"
+        :class="contentClass"
+        :aria-labelledby="`base-tabs-panel-${selectedTab}`"
+        data-test="base-tabs-panel-content"
+        role="tabpanel"
+      >
+        <!--
+          @slot Slot used to display the selected tab content.
+            @binding {string} tab - This content's tab name.
+            @binding {function} selectTab - Function to select a tab.
+            @binding {function} deselectTab - Function to deselect the tab.
+        -->
+        <slot
+          :name="selectedTab"
+          v-bind="{ tab: selectedTab, selectTab, deselectTab: () => selectTab(selectedTab) }"
+        />
+      </div>
+    </component>
+  </section>
+</template>
+
+<script lang="ts">
+  import Vue from 'vue';
+  import { mixins } from 'vue-class-component';
+  import { Component, Prop } from 'vue-property-decorator';
+  import { NoElement } from '../no-element';
+  import { dynamicPropsMixin } from '../dynamic-props.mixin';
+
+  /**
+   * Base Tabs Panel.
+   *
+   * @public
+   */
+  @Component
+  export default class BaseTabsPanel extends mixins(
+    dynamicPropsMixin(['activeTabClass', 'contentClass', 'tabClass', 'tabsListClass'])
+  ) {
+    /**
+     * Animation component that will be used to animate the tabs list.
+     *
+     * @public
+     */
+    @Prop({ default: 'header' })
+    public tabsAnimation!: Vue | string;
+
+    /**
+     * Animation component that will be used to animate the selected tab content.
+     *
+     * @public
+     */
+    @Prop({ default: () => NoElement })
+    public contentAnimation!: Vue | string;
+
+    /**
+     * The tab to be initially selected.
+     *
+     * @public
+     */
+    @Prop({ default: '' })
+    public initialTab!: string;
+
+    /**
+     * Allows the tabs to be unselected.
+     *
+     * @public
+     */
+    @Prop({ default: false, type: Boolean })
+    public allowTabDeselect!: boolean;
+
+    /**
+     * The currently selected tab.
+     *
+     * @internal
+     */
+    protected selectedTab: string = this.initialTab;
+
+    /**
+     * Extracts the tab from the slots.
+     *
+     * @returns The list of tabs.
+     *
+     * @internal
+     */
+    protected getTabs(): string[] {
+      return Object.keys(this.$scopedSlots).filter(
+        slotName => !['tab', 'tab-content'].includes(slotName)
+      );
+    }
+
+    /**
+     * Changes the current selected tab. If the tab is already selected
+     * and `allowTabDeselect` is `true`, the tab will be unselected.
+     *
+     * @param tab - The tab to be selected.
+     *
+     * @internal
+     */
+    protected selectTab(tab: string): void {
```

Comment:
As now we can select and desect I'd say that something like `toggleTab` would be easier to understand, maybe?

Reply:
It's a good point, I did not choose `toggleTab` as it is not always toggling the selection, it will only do it when `allowTabDeselect` is `true`.